metadata/metadata.go

```diff
@@ -270,16 +286,15 @@ func get(ctx context.Context, hang bool) (*Descriptor, error) {
 }
 
 // WriteGuestAttributes does a put call to mds changing a guest attribute value.
-func WriteGuestAttributes(key, value string) error {
+func (c *Client) WriteGuestAttributes(key, value string) error {
```

Comment:
since we're already updating this, can we also change its signature to accept context from caller and use it while making HTTP calls?

Reply:
surething. will do. Thanks for catching this one.

Reply:
done.